### PR TITLE
Us111 logout button

### PR DESCRIPTION
**Pull request by NathanJGaul**
### Overview

This PR implements the ability for a user to logout on the client side and makes a few small changes to the `/logout` api on the server side.

### Includes:

- client/src/main/App.tsx
- client/src/main/components/TopMenuBar.tsx
- client/src/main/logic/fetchMethods.ts
- node_server/src/main/routes/logout.ts

### Developer Checklist:

- [x] My code compiles and runs locally
- [x] The code follows the `QualityPolicy.md` file
- [x] My code has been developer-tested and includes unit tests
- [x] I have considered proper use of exceptions
- [x] I have eliminated IDE warnings

### Refs:

- [US111](https://tree.taiga.io/project/aneuhold-pointspire/us/111?milestone=268243)

### CODE REVIEW

**Comment by aneuhold** [comment]
When refreshing the page it seems to pull in the development user information instead of the currently logged in user's information, after the user has successfully logged in. Please see [this video for an example](https://youtu.be/sDVatb93ghc).

Also please see below for the `.env` file being used when getting these results :)

```
# All environment variables here should have `REACT_APP_` prepended to them
# besides EXTEND_ESLINT. That way react actually picks them up.

# REACT_APP_ENV can be set to LOCAL_DEV which will make it so the server called
# will be from localhost and a test user will be pulled. You can also use
# LOCAL to make it so the test user isn't used, but the server is still used
# on localhost:8055.
REACT_APP_ENV=LOCAL

# REACT_APP_AUTH can be set to LOCAL so that it uses a different Oauth method
# on github which sets the callback to localhost:3000 instead of the normal
# PointSpire site.
REACT_APP_AUTH=LOCAL

EXTEND_ESLINT=true
```

File: client/src/main/App.tsx [hunk 1]
```diff
@@ -306,6 +306,7 @@ class App extends React.Component<AppProps, AppState> {
             updateSettings={updateSettings}
             appTheme={appTheme}
             setTheme={setTheme}
+            loggedIn={!!user}
```
Comment (aneuhold) [inline client/src/main/App.tsx:309]: That is a clever use of the not operator. lol

File: node_server/src/main/routes/logout.ts [hunk 2]
```diff
@@ -12,11 +12,14 @@ export default function createLogoutRouter(): Router {
...
    */
   router.use('/', (req, res, next) => {
+    res.setHeader('Access-Control-Allow-Credentials', 'true');
     if (req.session) {
       req.session.destroy(err => {
         if (err) {
           next(err);
         } else {
+          res.clearCookie('connect.sid');
+          res.clearCookie('userId');
```
Comment (aneuhold) [inline node_server/src/main/routes/logout.ts:22]: Is there a way to clear all cookies? Or do we not want to do that here?

File: client/src/main/App.tsx (hunk 1 above)
Comment (NathanJGaul) [reply client/src/main/App.tsx:309]: wasn't my idea lol, eslint suggested this instead of `user ? true : false`

Comment (aneuhold) [reply client/src/main/App.tsx:309]: Eslint doing a good job here then haha nice

File: node_server/src/main/routes/logout.ts (hunk 2 above)
Comment (NathanJGaul) [reply node_server/src/main/routes/logout.ts:22]: Hmm, not an obvious way to clear all client side cookies from server (see https://expressjs.com/en/api.html). Might be able to do it from the client side.

Comment (aneuhold) [reply node_server/src/main/routes/logout.ts:22]: This seems to have been fixed for now.

**Pull request merged**